src/BotPlutusInterface/CardanoCLI.hs

```diff
@@ -197,17 +198,24 @@ buildTx ::
   forall (w :: Type) (effs :: [Type -> Type]).
   Member (PABEffect w) effs =>
   PABConfig ->
+  Map PubKeyHash DummyPrivKey ->
   PubKeyHash ->
   BuildMode ->
   Tx ->
   Eff effs ()
-buildTx pabConf ownPkh buildMode tx =
+buildTx pabConf privKeys ownPkh buildMode tx =
   callCommand @w $ ShellArgs "cardano-cli" opts (const ())
   where
     ownAddr = Ledger.pubKeyHashAddress (Ledger.PaymentPubKeyHash ownPkh) Nothing
+    skeys = Map.filter (\case FromSKey _ -> True; FromVKey _ -> False) privKeys
     requiredSigners =
       concatMap
-        (\pubKey -> ["--required-signer", signingKeyFilePath pabConf (Ledger.pubKeyHash pubKey)])
+        ( \pubKey ->
+            let pkh = Ledger.pubKeyHash pubKey
+             in if Map.member pkh skeys
+                  then ["--required-signer", signingKeyFilePath pabConf pkh]
+                  else ["--required-signer-hash", encodeByteString $ fromBuiltin $ getPubKeyHash pkh]
+        )
```

Comment:
Could perhaps save some computation here by loosing the filter, and changing the if statement to a case over `lookup`, matching `Just (FromSKey _) ...`

Reply:
Makes a lot more sense then my version